plugins/custom_caret.lua

```diff
@@ -106,4 +119,28 @@ function DocView:draw_caret(x, y)
   end
 
   renderer.draw_rect(x, y, caret_width, caret_height, caret_color)
+  if current_caret_shape == "block" then
+    core.push_clip_rect(x, y, caret_width, caret_height)
+
+    local function draw_char(l, c)
+      l = common.clamp(l, 1, #self.doc.lines   )
+      c = common.clamp(c, 1, #self.doc.lines[l])
+      local cx,cy = self:get_line_screen_position(l, c)
+      renderer.draw_text(
+        font, self.doc:get_char(l, c),
+        cx, cy+self:get_line_text_y_offset(),
+        style.background
+      )
+    end
+
+    for yo=-1, 1 do
+      for xo=-1, 1 do
+        draw_char(line+xo, col+yo)
+      end
+    end
```

Comment:
Why do you need to redraw other chars?

Reply:
it's just for when moving between characters, more of a cosmetic thing than anything else

Reply:
Hopefully it doesn't impact performance much. I'd add a comment about the reason tho.

Reply:
even better, I made it a setting
![Captura desde 2023-05-04 22-49-47](https://user-images.githubusercontent.com/70547062/236368604-f4155b32-c666-4ee2-9392-9ab0b2bc6a25.png)